src/Layout/Footer.tsx

```diff
@@ -239,21 +235,14 @@ function LanguageModal({
             <RadioGroup
               aria-label="language"
               name="language"
-              value={language}
+              value={i18n.language}
               onChange={(event) => {
-                setlanguage(event.target.value);
-                localStorage.setItem("language", event.target.value);
-                if (
-                  projectDetails &&
-                  projectDetails.purpose !== "planet-cash-signup" &&
-                  router.query.to
-                ) {
-                  loadPaymentSetup({
-                    projectGUID: router.query.to as string,
-                    paymentSetupCountry: country,
-                    shouldSetPaymentDetails: false,
-                  });
-                }
+                const lang = event.target.value;
+                const { pathname, query, asPath } = router;
+                // set cookie to store user preferences
+                document.cookie = `NEXT_LOCALE=${lang}; max-age=31536000; path=/`;
+                // change just the locale and maintain all other route information including href's query
+                router.push({ pathname, query }, asPath, { locale: lang });
```

Comment:
This push unfortunately currently can lead to showing the wrong page if switching the language on the contact details page (e.g. http://localhost:3000/?to=donateatreeelmagistral&step=contact) by showing the donation options screen again and resulting into the "continue" button not to work any more and the user got stuck:

<img width="743" alt="Bildschirm­foto 2023-03-18 um 13 19 29" src="https://user-images.githubusercontent.com/1532418/226105617-915e35cc-3fdc-4951-a5e4-48fd34be9e22.png">

Reply:
I'll disallow language change if the user is on the Contact Form or the Payment Form for now - it is unlikely that the user would want to switch at that step.

 Will reconsider whether to allow that while refactoring the donation code.

Reply:
Fixed in 8e6b9a35406b1aa9526e719464821292a30e9793